File: backend/dalathasfarm/src/main/java/com/example/dalathasfarm/models/OrderDetail.java
package com.example.dalathasfarm.models;

import jakarta.persistence.*;
import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.Min;
import lombok.*;

import java.math.BigDecimal;

@Entity
@Table(name = "order_details")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class OrderDetail {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "order_id", nullable = false)
    private Order order;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;

    @Column(nullable = false)
    @Min(1)
    private Integer quantity;

    @Column(nullable = false, precision = 10, scale = 2)
    @DecimalMin(value = "0.0", inclusive = true)
    private BigDecimal price;

    @Column(name = "total_money",nullable = false, precision = 10, scale = 2)
    @DecimalMin(value = "0.0", inclusive = true)
    private BigDecimal totalMoney;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "coupon_id")
    private Coupon coupon;
}

File: android/lib/models/order_detail.dart
import 'package:android/models/order.dart';
import 'package:android/models/product.dart';
import 'package:decimal/decimal.dart';

class OrderDetail {
    final int id;
    final Order order;
    final Product product;
    final Decimal price;
    final String quantity;
    final Decimal totalPrice;

    OrderDetail({
        required this.id,
        required this.order,
        required this.product,
        required this.price,
        required this.quantity,
        required this.totalPrice,
    });

    factory OrderDetail.fromJson(Map<String, dynamic> json) {
        return OrderDetail(
            id: json['id'],
            order: Order.fromJson(json['order']),
            product: Product.fromJson(json['product']),
            price: json['price'],
            quantity: json['quantity'],
            totalPrice: json['totalPrice'],
        );
    }
}